addons/stock/models/stock_quant.py

```diff
@@ -340,20 +338,58 @@ def get_import_templates(self):
         }]
 
     @api.model
-    def _get_forbidden_fields_write(self):
-        """ Returns a list of fields user can't edit when he want to edit a quant in `inventory_mode`."""
-        return ['product_id', 'location_id', 'lot_id', 'package_id', 'owner_id']
+    def _trigger_inventory_moves(vals):
+        return any(field in vals for field in ['location_id', 'lot_id', 'package_id', 'owner_id'])
+
+    @api.model
+    def _get_specific_fields_write(self):
+        """ Model method.
+        Return a list of field for which a balanced pair of move are created on edition.
+        :return: List of String
+        """
+        return ['location_id', 'lot_id', 'package_id', 'owner_id']
+
+    def _update_quant_by_moves(self, vals):
+        """ Handle update of more specific fields"""
+        # if any key of vals in specific fields => Move+ & Move- with all values
+
+        move_vals = list(chain.from_iterable([
+            # 'Undo' actual quant
+            quant._get_inventory_move_values(
```

Comment:
I would make it a little smarter for `location` (one move is enough) and for package (one move line). You can check `move_quant()` implementation